connectd/connectd.c

```diff
@@ -976,8 +983,43 @@ static void try_connect_one_addr(struct connecting *connect)
 			af = AF_INET6;
 			break;
 		case ADDR_TYPE_DNS:
-			// TODO: resolve with getaddrinfo and set af
-			break;
+			/* Resolve with getaddrinfo */
+			memset(&hints, 0, sizeof(hints));
+			hints.ai_socktype = SOCK_STREAM;
+			hints.ai_family = AF_UNSPEC;
+			hints.ai_protocol = 0;
+			hints.ai_flags = AI_ADDRCONFIG;
+			gai_err = getaddrinfo((char *)addr->u.wireaddr.addr,
+					      tal_fmt(tmpctx, "%d",
+						      addr->u.wireaddr.port),
+					      &hints, &ais);
+			if (gai_err != 0)
+				status_failed(STATUS_FAIL_INTERNAL_ERROR,
+					      "DNS with getaddrinfo gave: %s",
+					      gai_strerror(gai_err));
```

Comment:
Err, no!  If it doesn't resolve for some reason, don't shutdown lightningd!  Just skip it (maybe LOG_DEBUG the error?).

Reply:
Whops sure, fixed this. Now I use `status_debug` and `goto next;`. Question is if I need to do a `freeaddrinfo(ais)` when we encountered a `gai_error` from `getaddrinfo`. I think not, because that would be bad design.